### PR TITLE
forward addExtension call to inner factory

### LINKED ISSUE #249: MenuBundle disable all the knp menu extensions
Menu Bundle disabled all the knp menu extensions when knp menu bundle regist after the cmf menu bundle，because the knp menu extension pass can not find the knp_menu.factory service

```php
class AddExtensionsPass implements CompilerPassInterface
{
    public function process(ContainerBuilder $container)
    {   
        if (!$container->hasDefinition('knp_menu.factory')) {
            return;
        }   

        $definition = $container->getDefinition('knp_menu.factory');

        foreach ($container->findTaggedServiceIds('knp_menu.factory_extension') as $id => $tags) {
            foreach ($tags as $tag) {
                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
                $definition->addMethodCall('addExtension', array(new Reference($id), $priority));
            }   
        }   
    }   
}
```

### PR DESCRIPTION
**Pull request by dbu**
fix #249, replace #250.

also needs https://github.com/KnpLabs/KnpMenuBundle/pull/317 to actually work.

### CODE REVIEW

**Comment by efeencheung** [comment]
By use has and findDefinition，and add this commit to my local code，the extension works now，thank @dbu

File: QuietFactory.php [hunk 1]
```diff
@@ -69,10 +70,28 @@ public function createItem($name, array $options = array())
...
+     *
+     * @param ExtensionInterface $extension
+     * @param int                $priority
+     *
+     * @throws \Exception if the inner factory does not implement the addExtension method.
+     */
+    public function addExtension(ExtensionInterface $extension, $priority = 0)
+    {
+        if (!method_exists($this->innerFactory, 'addExtension')) {
+            throw new \Exception(sprintf(
```
Comment (WouterJ) [inline QuietFactory.php]: I would use a more specific `LogicException` here

**Comment by WouterJ** [comment]
:+1:

**Comment by dbu** [comment]
updated the exception type

**Comment by WouterJ** [comment]
What about cherry-picking this into the 2.0 branch and releasing a 2.0.1 version?

**Comment by dbu** [comment]
good point. see #260

**Pull request merged**